core/src/gossip_service.rs

```diff
@@ -138,13 +138,24 @@ pub fn get_multi_client(nodes: &[ContactInfo]) -> (ThinClient, usize) {
         .iter()
         .filter_map(ContactInfo::valid_client_facing_addr)
         .map(|addrs| addrs)
+        .filter(|addrs| {
```

Comment:
Might be a bit slow to do this if there is a large list of potential nodes.

Reply:
Maybe 1s per node if they're responding well, nbd in the grander scheme of things.

Reply:
Well in certain cases if you had 200-300 nodes, then 200-300 seconds or 3-5 minutes might be painful to wait for one command to complete, but I'm fine to try it out for now.

Reply:
The multi-client should filter these out eventually once it gets timed-out for `.transaction_count()`  and/or `.transfer()`

Reply:
Oh where's that?  iirc I was seeing failures due to being unable to fetch a recent blockhash from one of these bad RPC nodes.  I didn't record the exact location of the failure though

Reply:
It was probably here: https://github.com/solana-labs/solana/blob/9f354522a745e54016ba514b54ab9c2d529e22ee/bench-tps/src/bench.rs#L513

Reply:
Maybe we should either not `experiment` at:
https://github.com/solana-labs/solana/blob/f551b34725d0eb8691991b40e27b160322512166/client/src/thin_client.rs#L339
and 
https://github.com/solana-labs/solana/blob/f551b34725d0eb8691991b40e27b160322512166/client/src/thin_client.rs#L381,
or on failure we continue the `experiment` until we find a node that works or we run out of nodes?

Reply:
Yea. Generally I was going with the pattern that the client should observe the failure and then do what it wants to retry or not.

https://github.com/solana-labs/solana/blob/0cbc0dc79c7c5e429d26ff336fbd6b5c5d8b1329/client/src/perf_utils.rs#L40

Kind of like that case. I think either way could work if the thin_client itself wants to keep cycling through nodes until as you said it runs out.

Reply:
WDYT about https://github.com/solana-labs/solana/pull/5813?  Better than this PR?

Reply:
I think so, yea.